docs/versions/unversioned/sdk/google-sign-in.md

```diff
@@ -0,0 +1,424 @@
+
+## Setup
+
+For questions on setup, feel free to comment on this post: [**React Native Google Sign-Up**](https://blog.expo.io/react-native-google-sign-in-with-expo-d1707579a7ce)
+
+1. Go to your `app.json` and make sure you define your `ios.bundleIdentifier` and the `android.package` you want to use.
+2. Open up [the Firebase Console](https://console.firebase.google.com) and setup a new project, or use an existing one.
+3. Create a native iOS, and Android app using the Bundle ID and Android package you defined earlier.
+4. Download the `GoogleService-info.plist` (iOS) & the `google-services.json` (Android). Move them to your Expo project.
+5. In the `app.json`, set your `expo.ios.config.googleSignIn.reservedClientId` to the value of `REVERSE_CLIENT_ID` in the `GoogleService-info.plist`.
+6. Also in `app.json`, set `expo.android.googleServicesFile` to the relative path of your `google-services.json`. Make sure the file is located somewhere in your Expo project.
+
+```js
+ // app.json
+{
+  "expo": {
+    "ios": {
+      // The bundle ID you used with your Firebase app
+      "bundleIdentifier": "example.expo.googlesignin",
+      "config": {
+        "googleSignIn": {
+          // Your REVERSE_CLIENT_ID from the GoogleService-info.plist
+          "reservedClientId": "<YOUR_IOS_CLIENT_ID>"
+        }
+      }
+    },
+    "android": {
+        // The package you used with your Firebase app
+        "package": "example.expo.googlesignin",
+        // Relative path to the file generated by Firebase
+        "googleServicesFile": "./google-services.json"
+    }
+  }
+}
+```
+
+At this point you can build your project and upload it to the App Store or the Google Play Store.
+
+When your app is built you can verify that the iOS URL Scheme is properly set up by reading it using the `expo-app-auth` module in a standalone app.
+
+```js
+import { AppAuth } from 'expo-app-auth';
+
+// This value should contain your REVERSE_CLIENT_ID
+const { URLSchemes } = AppAuth;
+```
+
+## Initialize the API
+
+Before using the API we first need to call `GoogleSignIn.initAsync({ ... })` which configures how sign-in functionality will work.
+
+```js
+try {
+  await GoogleSignIn.initAsync({ clientId: '<YOUR_IOS_CLIENT_ID>' });
+} catch ({ message }) {
+  alert('GoogleSignIn.initAsync(): ' + message);
+}
+```
+
+## Signing in/out
+
+```js
+signInAsync = async () => {
+  try {
+    await GoogleSignIn.askForPlayServicesAsync();
+    const { type, user } = await GoogleSignIn.signInAsync();
+    if (type === 'success') {
+      // ...
+    }
+  } catch ({ message }) {
+    alert('login: Error:' + message);
+  }
+};
+```
+
+```js
+signOutAsync = async () => {
+  try {
+    await GoogleSignIn.signOutAsync();
+    this.setState({ user: null });
+  } catch ({ message }) {
+    alert('signOutAsync: ' + message);
+  }
+};
+```
+
+## Methods
+
+### `getPlayServiceAvailability(shouldAsk: boolean = false): Promise<boolean>`
```

Comment:
```suggestion
### `getPlayServiceAvailabilityAsync(shouldAsk: boolean = false): Promise<boolean>`
```

Also not a fan of the boolean argument -- without reading the docs recently, what does `getPlayServiceAvailability(false)` do? Boolean args are nonsensical unless the method name explains what they do (`setEnabled(false)` is clear). Splitting this into two functions like `getPlayServiceAvailabilityAsync(void)` and `askForPlayServices(void)` is clearer to readers of the code.